src/main/java/com/github/shyiko/mysql/binlog/MariadbBinaryLogClient.java

```diff
@@ -0,0 +1,94 @@
+package com.github.shyiko.mysql.binlog;
+
+import com.github.shyiko.mysql.binlog.event.*;
+import com.github.shyiko.mysql.binlog.event.deserialization.*;
+import com.github.shyiko.mysql.binlog.network.protocol.command.Command;
+import com.github.shyiko.mysql.binlog.network.protocol.command.DumpBinaryLogCommand;
+import com.github.shyiko.mysql.binlog.network.protocol.command.QueryCommand;
+
+import java.io.IOException;
+
+/**
+ * Mariadb replication stream client.
+ *
+ * @author <a href="mailto:winger2049@gmail.com">Winger</a>
+ */
+public class MariadbBinaryLogClient extends BinaryLogClient {
+
+    private boolean useSendAnnotateRowsEvent;
+
+    public MariadbBinaryLogClient(String username, String password) {
+        super(username, password);
+    }
+
+    public MariadbBinaryLogClient(String schema, String username, String password) {
+        super(schema, username, password);
+    }
+
+    public MariadbBinaryLogClient(String hostname, int port, String username, String password) {
+        super(hostname, port, username, password);
+    }
+
+    public MariadbBinaryLogClient(String hostname, int port, String schema, String username, String password) {
+        super(hostname, port, schema, username, password);
+    }
+
+    @Override
+    protected GtidSet newGtidSet(String gtidSet) {
+        return new MariadbGtidSet(gtidSet);
+    }
+
+    @Override
+    protected void setupGtidToPurgedIfNeeded() throws IOException {
+        //Mariadb ignore
```

Comment:
does maria-db not support the `gtid_purged` variable?

Reply:
yeah, u can find this in https://mariadb.com/kb/en/system-variable-differences-between-mariadb-100-and-mysql-56/